### DISCUSSION TITLE
How do you handle SFP modules?

### DISCUSSION THREAD

**Discussion by lucker7777** [netbox-community/netbox#5722] [Q&A] [answered]
Hello,
My job is to migrate a huge network infrastructure from the old proprietary software to Netbox.
I am struggling with SFP modules as we have over 9000 SFP modules, each with unique ID, each connected with fiber.

I tried to read as much Netbox documentation and discussions as possible, but so far I was unable to find an appropriate solution.
I found out that Devices have Interfaces, which might be a type SFP. But this is just an interface/port, not module. Module can be removed, or even can be separated from the switch.
Furthermore, I found out, that Devices can have Inventory Items, but those cannot be connected by cables.
Finally, there are Device Bays, but I do not feel like those are appropriate for SFP modules.

Do you have any ideas or experience with this?

Thanks :)

**Comment by candlerb** [comment, marked as answer]
If you want to record SFP modules in Netbox, then they would go in as Inventory Items on a Device.  This lets you record the manufacturer, model and serial number.

Points to note:

* There is no explicit association between a particular Inventory Item and the interface into which it's plugged.  You'd have to do that via the naming scheme for the SFP (e.g. name it after the interface), or its description.
* There's no way in Netbox to track inventory items which are not installed in any device.  Indeed, you can't even move an inventory item from one device to another; you have to delete it from one and re-add it to the other.  If you need a spares inventory, then you'll need a separate asset management system.
* In Netbox, the fibre goes into the Interface, not into the SFP module.
* Netbox does not model individual fibre strands.  If your SFP has separate transmit and receive interfaces and takes a duplex cable, this is still one "cable" going into one "interface" as far as Netbox is concerned.

You are correct that device bays are not appropriate for SFP modules.  Device bays are for independently managed devices which just happen to sit inside another device.  Blade servers in a blade chassis are appropriate use of device bays; but not line cards inside a chassis switch (because it's a single managed entity).

**Comment by lucker7777** [comment]
Thanks for clarification @candlerb.
Do you know any problems I might run into if I abuse Device Bays for SFPs?
Is it possible that some Netbox users handle SFPs like that?

**Reply by candlerb** [reply]
> Do you know any problems I might run into if I abuse Device Bays for SFPs?

The Netbox Gods will smite you :-)

Seriously: every single SFP you own will appear as a "Device".  They will appear in lists of devices alongside real devices.  The data model will suggest to users that each SFP can have interfaces and IP addresses and Role and Platform.  You don't want this.

Worse, if you give an SFP "device" its own "interface", accepting a cable here won't work: this interface belongs to the SFP as if it were a standalone server, not to the parent device (which is where you want it to go).  The cable *must* go directly into the parent device.

Given that's the case, there's no reason to use child Device rather than Inventory Item.  Inventory Item records the key things that you want for an SFP: the manufacturer, model, serial number, asset tag, and name/label/description.  You can use one of these latter ones to associate it with an interface on the Device.

**Reply by lucker7777** [reply]
I do not want to mess with the Netbox Gods :)

Thanks so much for help.

**Comment by Hexdelete** [comment]
https://netboxlabs.com/docs/netbox/models/dcim/inventoryitem/

FYI - Inventory Items are being deprecated and they suggest using modules.

**Reply by TatersShip** [reply]
[https://netboxlabs.com/docs/netbox/models/dcim/inventoryitem/](url)

Here is the public link, thanks for saying something as I was about to start this process myself.

**Reply by Hexdelete** [reply]
What a n00b mistake on my part! Thank you for the proper link!!
